### ISSUE TITLE
unable to comment on PR with github token 

### ISSUE THREAD

**Issue by ilyesAj** [dflook/terraform-github-actions#276]
### Problem description

I'm trying to use GitHub runner's token to publish a message on a PR but I'm facing an error of `username not found`. 

is related to [this comment](https://github.com/dflook/terraform-github-actions/blob/main/image/src/github_pr_comment/__main__.py#L241) mentioning that not all tokens can be used? 

### Backend

s3

### Workflow YAML

```yaml
name: PR Plan

on: [pull_request]

permissions:
  contents: read
  pull-requests: write

jobs:
  plan:
    runs-on: [self-hosted,docker]
    name: Create terraform plan
    env:
      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }} 
    steps:
      - name: Checkout
        uses: actions/checkout@v3

      - name: terraform plan
        uses: dflook/terraform-plan@v1
        with:
            #var_file: dev.tfvars
            backend_config: access_key=${{ secrets.xxx }} secret_key=${{ secrets.xxx }}
```

### Workflow log

```shell
Plan: 1 to add, 0 to change, 1 to destroy.
805
##[debug]['/usr/local/bin/github_pr_comment', 'plan']
806
##[debug]Backend fingerprint includes dict_keys(['endpoint', 'bucket', 'workspace_key_prefix', 'key'])
807
##[debug]Backend fingerprint includes dict_keys(['endpoint', 'bucket', 'workspace_key_prefix', 'key'])
808
##[debug]Wrote pr_url to step_cache
809
##[debug]GET https://ghe.xxx.com/api/v3/repos/sre-eu/terraform-cicd/pulls/1 -> 200
810
##[debug]Wrote issue-url/55be76b7dbbc1dc0adbbbe8ce5fe45470e764b64686916b7c17530c4940927e1 to job_cache
811
##[debug]POST https://ghe.xxx.com/api/v3/graphql -> 404
812
##[debug]{"message":"Not Found","documentation_url":"https://docs.github.com/enterprise-server@3.8/rest"}
813
##[debug]Failed to get current user from graphql
814
##[debug]GET https://ghe.xxx.com/api/v3/user -> 403
815
Not Found
816
Traceback (most recent call last):
817
  File "/usr/local/bin/github_pr_comment", line 8, in <module>
818
    sys.exit(main())
819
  File "/usr/local/lib/python3.9/dist-packages/github_pr_comment/__main__.py", line 386, in main
820
    comment = get_comment(action_inputs, backend_fingerprint, partial_backend_fingerprint)
821
  File "/usr/local/lib/python3.9/dist-packages/github_pr_comment/__main__.py", line 291, in get_comment
822
    username = current_user(env)
823
  File "/usr/local/lib/python3.9/dist-packages/github_pr_comment/__main__.py", line 248, in current_user
824
    raise Exception('Unable to get username for the github token')
825
Exception: Unable to get username for the github token
826
##[debug]ls -la /github/workspace:total 44
```

**Comment by DanyC97** [comment]
@ilyesAj  i think what you see is not a problem of the action itself.  I'd suggest you either use a PAT or a token generated with the help of GH APP.

**Comment by ilyesAj** [comment]
PAT works fine but i wanted to have github action bot like this using github token :
<img width="959" alt="Screenshot 2023-07-13 at 18 48 29" src="https://github.com/dflook/terraform-github-actions/assets/34598860/5f647db3-30a0-40bf-9e24-ec13be182869">

this done by the following action: [EnricoMi/publish-unit-test-result-action@v2](https://github.com/EnricoMi/publish-unit-test-result-action)
is it possible to make it with terraform-plan workflow ?

**Comment by jrspinella** [comment]
I got the same error on GHES.

**Comment by DanyC97** [comment]
> PAT works fine but i wanted to have github action bot like this using github token : <img alt="Screenshot 2023-07-13 at 18 48 29" width="959" src="https://user-images.githubusercontent.com/34598860/253350409-5f647db3-30a0-40bf-9e24-ec13be182869.png">
> 
> this done by the following action: [EnricoMi/publish-unit-test-result-action@v2](https://github.com/EnricoMi/publish-unit-test-result-action)

imo the mentioned action does support what i mentioned - see https://github.com/EnricoMi/publish-unit-test-result-action/blob/master/action.yml#L6-L10 and i'd argue that is the right way to achieve what you need.



>  is it possible to make it with terraform-plan workflow ?

i'm not the maintainer however by view is that would be wrong. The set of actions created by @dflook  are generic and well scoped, would be a mistake to "enhance" and follow a path which is not the right one. What you want to achieve can be done with a GH App.

**Comment by DanyC97** [comment]
> I got the same error on GHES.

not sure i understand the issue. Are you saying that

* you trying to use GITH?UB_TOKEN (default one) to post a comment AND you would like the actor to be shown as github-actions?
* or you are unable to post a comment in GHES using the default GITHUB_TOKEN ?

**Comment by dflook** [comment]
Thanks for creating an issue @ilyesAj. 

Any type of token should work with github.com, but it looks like youyr using GHES which may work differently (and I don't test with). 

It's not clear where the GITHUB_TOKEN provided by the runner has come from. The screenshot shows a comment by an app called 'github-actions', but it doesn't have the profile picture that the github-actions bot on github.com uses. Perhaps this is normal for GHES? Or is this a github app you've created yourself?

What type of runner are you using? Are you using [the offical runner](https://github.com/actions/runner) that's been manually added to the repo/org, or [something else](https://jonico.github.io/awesome-runners/)?

**Comment by dflook** [comment]
I can't test it, but this may be fixed in v1.36.1. Could you give it a try?

**Comment by ilyesAj** [comment]
> I can't test it, but this may be fixed in v1.36.1. Could you give it a try?

thank @dflook i will make a try

**Comment by dflook** [comment]
I believe this is fixed in v1.36.2. If you have any more problems please create a new issue.

**Comment by ilyesAj** [comment]
i just tested it yesterday , it's working like a charm thank you !
